### PR TITLE
Add data generation job samples to azure-ai-projects

### PR DESCRIPTION
**Pull request by aprilk-ms**
## What

Adds four new samples under `sdk/ai/azure-ai-projects/samples/datasets/` that cover the four hero scenarios of the 2.2.0a data generation API (`pc.beta.datasets.create_generation_job`), plus a README index.

| File | Source(s) | Scenario | Demonstrates |
|---|---|---|---|
| `sample_dataset_generation_job_traces_for_evaluation.py` | Traces | Evaluation | Generate an eval dataset from a Foundry agent's recent App Insights traces. |
| `sample_dataset_generation_job_traces_for_finetuning.py` | Traces | Supervised fine-tuning | Same source, with `train_split=0.8` producing two JSONL file outputs. Resolves the real filename of each Azure OpenAI file output. |
| `sample_dataset_generation_job_simpleqna_with_dataset_source.py` | Dataset + Prompt | Evaluation | Multi-source: combines a seed Dataset with an inline Prompt that steers question difficulty; shows that caller-supplied `description` and `tags` propagate onto the generated dataset. |
| `sample_dataset_generation_job_simpleqna_for_finetuning.py` | File (Azure OpenAI) | Supervised fine-tuning | Uploads a reference document via the Azure OpenAI Files API, waits for it to reach `processed` status, then asks for both short-answer and long-answer fine-tuning files. |

Also adds `sdk/ai/azure-ai-projects/samples/datasets/README.md` mirroring `samples/evaluations/README.md`.

## Style

Matches the existing `sample_dataset_generation_job_with_evaluation.py` reference exactly: MIT header, parenthesized multi-`with`, dot-style polling, `TERMINAL_STATUSES` set, deletes the data generation job at the end but leaves the generated output (dataset or files) for the user. The SimpleQnA samples additionally delete the temporary inputs they upload (seed dataset / Azure OpenAI file).

## Service quirks handled in the samples

* All four preflight `len(output_name) <= 50` because the service rejects longer output names with a 400 `invalidPayload`.
* Sample #4 polls the uploaded Azure OpenAI file until `status == processed` before submitting the job (otherwise the service returns `must point to a completed file import`).
* Sample #4's seed reference document is >1 KB because the service rejects smaller files with `File content is too small to generate QnA. It must be at least 1KB`.
* Samples #2 and #4 call `openai_client.files.retrieve(output.id).filename` because the service returns `output.filename = None` on FT outputs today.

## Validation

All four samples were live-validated end-to-end against a real Foundry project (`foundryljm7/projectljm7` with `test-agent` and `gpt-4.1`). Each produced the expected dataset / training+validation file pair, and cleaned up its temporary inputs successfully.

## Target branch

This PR targets `feature/azure-ai-projects/2.2.0` (NOT main).

### CODE REVIEW

File: sdk/ai/azure-ai-projects/samples/datasets/README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+```bash
+pip install "azure-ai-projects>=2.0.0" azure-identity python-dotenv
+```
+
+To run asynchronous samples, you will also need to install `aiohttp`. The data generation samples that interact with Azure OpenAI files (any sample that emits or consumes Azure OpenAI File outputs/inputs) also require `openai`.
+
+Set these environment variables:
+- `FOUNDRY_PROJECT_ENDPOINT` - Required for all samples. Your Azure AI Project endpoint (e.g., `https://<account>.services.ai.azure.com/api/projects/<project>`).
+- `FOUNDRY_MODEL_NAME` - Required for `simple_qna` data generation samples (`sample_dataset_generation_job_with_evaluation.py`, `sample_dataset_generation_job_simpleqna_with_dataset_source.py`, `sample_dataset_generation_job_simpleqna_for_finetuning.py`). The model deployment name (e.g., `gpt-4o-mini`).
+- `FOUNDRY_AGENT_NAME` - Required for traces-based data generation samples (`sample_dataset_generation_job_traces_for_evaluation.py`, `sample_dataset_generation_job_traces_for_finetuning.py`). The name of a Foundry agent with recent traces in Application Insights.
```
Comment (sakoll) [inline sdk/ai/azure-ai-projects/samples/datasets/README.md]: Actually we support Foundry Agent Name even for eval SimpleQnA recipe. Also better to add one sample about it

File: sdk/ai/azure-ai-projects/samples/datasets/README.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+Most samples accept additional optional environment variables (`DATASET_NAME`, `POLL_INTERVAL_SECONDS`, `FOUNDRY_TRACES_WINDOW_DAYS`, etc.) — see each sample's docstring for details.
+
+## Running a Sample
+
+```bash
+# Set environment variables
+export FOUNDRY_PROJECT_ENDPOINT="https://<your-account>.services.ai.azure.com/api/projects/<your-project>"
+export FOUNDRY_MODEL_NAME="gpt-4o-mini" # Replace with your model
```
Comment (sakoll) [inline sdk/ai/azure-ai-projects/samples/datasets/README.md]: Should we call it `FOUNDRY_MODEL_DEPLOYMENT_NAME`. Also for eval simplqna, we especially need model which supports Responses API and it is better to link this page for supported models
https://learn.microsoft.com/en-us/azure/foundry/openai/how-to/responses?tabs=python-key#model-support
For Finetuning SimpleQnA, they need model which supports chat completions

File: sdk/ai/azure-ai-projects/samples/datasets/README.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+| Sample | Description |
+|--------|-------------|
+| [sample_datasets.py](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/ai/azure-ai-projects/samples/datasets/sample_datasets.py) | Upload files, create, list, and delete versioned Datasets |
+| [sample_datasets_async.py](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/ai/azure-ai-projects/samples/datasets/sample_datasets_async.py) | Async version of the dataset CRUD sample |
+| [sample_datasets_download.py](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/ai/azure-ai-projects/samples/datasets/sample_datasets_download.py) | Upload a folder as a Dataset and download its files via an Azure storage ContainerClient |
+
+### Data Generation Jobs
+
+Data generation jobs synthesize evaluation datasets or supervised fine-tuning files from different kinds of sources (a Foundry agent's traces, an inline prompt, an existing Dataset, or an Azure OpenAI File). The job runs server-side; the samples below show how to submit a job, poll it to completion, and locate the generated artifacts.
```
Comment (sakoll) [inline sdk/ai/azure-ai-projects/samples/datasets/README.md]: For traces, we also support 3P agents as well. So should we also mention  about it?

Comment (sakoll) [reply sdk/ai/azure-ai-projects/samples/datasets/README.md]: Similarly we support Agent source for eval simpleqna

File: sdk/ai/azure-ai-projects/samples/datasets/sample_dataset_generation_job_simpleqna_for_finetuning.py [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+        print(f"Generated samples: {job.result.generated_samples}")
+
+    # ------------------------------------------------------------------
+    # 4. Clean up.
+    # ------------------------------------------------------------------
+    print(f"Delete the Azure OpenAI input file `{seed_file.id}`.")
+    openai_client.files.delete(file_id=seed_file.id)
+
+    print(f"Delete the data generation job `{job.id}`.")
+    project_client.beta.datasets.delete_generation_job(job_id=job.id)
```
Comment (sakoll) [inline sdk/ai/azure-ai-projects/samples/datasets/sample_dataset_generation_job_simpleqna_for_finetuning.py:226]: Should we also delete generated dataset/AOAI file?

File: sdk/ai/azure-ai-projects/samples/datasets/sample_dataset_generation_job_simpleqna_with_dataset_source.py [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+
+    # ------------------------------------------------------------------
+    # 1. Upload the seed reference document as a versioned Dataset.
+    # ------------------------------------------------------------------
+    print(f"Upload the seed reference document as dataset `{seed_dataset_name}` v1.")
+    with tempfile.NamedTemporaryFile("w", suffix=".md", delete=False, encoding="utf-8") as tmp:
+        tmp.write(SEED_REFERENCE_DOCUMENT)
+        seed_local_path = tmp.name
+    try:
+        seed_dataset = project_client.datasets.upload_file(
```
Comment (sakoll) [inline sdk/ai/azure-ai-projects/samples/datasets/sample_dataset_generation_job_simpleqna_with_dataset_source.py]: This should be changed to AOAI File upload (similar to Finetuning), we don't support reading dataset input files.

File: sdk/ai/azure-ai-projects/samples/datasets/README.md (hunk 2 above)
Comment (aprilk-ms) [reply sdk/ai/azure-ai-projects/samples/datasets/README.md]: Added link. But I am keeping just model name since all of the samples use the same env name. I don't want to deviate from it

**Review by shivamMg — APPROVED** [review]

**Review by sakoll — APPROVED** [review]
